_discussion_reason_codes.md

```diff
@@ -0,0 +1,108 @@
+# Service Bus: reason codes or equivalent
+
+## Proposal
+
+Each language should provide:
+
+- [ ] Aconceptual equivalent to a reason code (a programatically readable field describing the reason for the failure)
+- [ ] A sample that shows how to use it.
+
+## Current state
+
+This is what we have:
+
+| Language   | Error class name                          | Has 'reason'                              | Samples link                                             |
+| ---------- | ----------------------------------------- | ----------------------------------------- | -------------------------------------------------------- |
+| JavaScript | [ServiceBusError][js_exception]           | yes                                       | [receiveMessagesStreaming.ts][js_sample]                 |
+| C#         | [ServiceBusException][csharp_exception]   | yes                                       | [ServiceBusProcessor(maybe, but not yet)][csharp_sample] |
+| Python     | [ServiceBusError(base)][python_exception] | yes, via the type of the exception itself |                                                          |
+| Java       | [ServiceBusException][java_exception]     | no                                        |                                                          |
+
+## Additional questions
+
+- [ ] Are users expected to react to these errors? If so, what other flags/information might we need?
```

Comment:
The attributes that I see as important are:

- `IsTransient` :  Offers a hint to users if retrying may resolve the issue;  also offers a hint to understand if the client has already retried according to its policy.

- `Message`:  What in the wide, wide world of sports is a-goin' on around here?   _(This is driven by the service response, when triggered as part of communication)_

One that I've heard asked for and discussed that I think we should avoid:

`WhoseFaultIsThis`:  An indication of "this was caused by something the user did"  versus  "this is just something that happened".      While it is potentially helpful, the service does not give a clear indication of this and I do not believe the client should be attempting to interpret and classify the service response this way.   If the service begins to return this, then yeah.... we should surface it.

Reply:
For the `isTransient`, how would we know if it is a transient issue if we pass on the error as soon as it is encountered?

Reply:
If you do not retry, it's not transient.   If you do....    😄

Reply:
We do have the retryable flag for that! :)

Reply:
Retryable = IsTransient

Reply:
Don't you mean `Retriable`? 😉